charmcraft/commands/store/store.py

```diff
@@ -315,6 +315,7 @@ def _upload(self, endpoint, filepath, *, extra_fields=None):
     def upload(self, name, filepath):
         """Upload the content of filepath to the indicated charm."""
         endpoint = f"/v1/charm/{name}/revisions"
+        self.whoami()
```

Comment:
I think this should go at the start of `_upload()`, because it looks like `upload_resource()`, which also calls `_upload()`, would also benefit from this change.
And then please add a comment to remind us  that the whoami call is to verify that the user is logged in; otherwise we might be tempted to remove this "spurious" call in the future.

Reply:
OK, will add.